apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerTransaction.cs

```diff
@@ -126,6 +127,24 @@ internal bool HasMutations
             }
         }
 
+        // Request options for the commit request of this transaction. Only the relevant properties are exposed publicly.
+        internal RequestOptions _commitRequestOptions = RequestOptions.Empty;
+
+        /// <summary>
+        /// The transaction tag to use for this transaction. This can only be set for read/write transactions,
+        /// and must be set before any statements are executed on the transaction.
+        /// </summary>
+        public string TransactionTag
+        {
+            get => _commitRequestOptions.TransactionTag;
+            set
+            {
+                GaxPreconditions.CheckState(Mode != TransactionMode.ReadOnly, "Transaction tag cannot be set on a read-only transaction");
```

Comment:
I don't think there's a test for this check... it would be useful as documentation.